src/command/backup/backup.c

```diff
@@ -365,7 +365,7 @@ backupBuildIncrPrior(const InfoBackup *infoBackup)
     Manifest *result = NULL;
 
     // No incremental if backup type is full
-    BackupType type = backupType(cfgOptionStr(cfgOptType));
+    BackupType type = (BackupType)cfgOptionStrId(cfgOptType);
```

Comment:
> The big question for me here is why are we not declaring type as `StringId type`

Actually, I'd say it should be the instances in load.c and expire.c that are converted to BackupType, which I have done in 67f35a88ca.

It's interesting that the switch statement did not complain that the type was not enum, even though the cases are enum. I'm not excited about just declaring everything as StringId because the enums provide some documentation and type safety in the cases where we are using the enum values.

> We sometimes comment why we're doing a cast when not obvious so maybe we should do that here

Integer casts are common enough that I'd rather not comment them all. And these are the type of casts that at least some compilers will let us know about when they are missing.

> so maybe something in stringid.h

I like this idea better. How about `When assigning a StringId to an enum, it will be necessary to cast the StringId to the enum type if the enum contains all 32-bit values, since some compilers will complain about the implicit conversion without a cast. The enum will be 32-bit if all values of the enum are <= 0xffffffff.`

I not sure it I want to go with string length because some 6 character 5-bit strings would be 32-bit. In fact about 1/4 of them would be.

Reply:
> > The big question for me here is why are we not declaring type as `StringId type`
> 
> Actually, I'd say it should be the instances in load.c and expire.c that are converted to BackupType, which I have done in [67f35a8](https://github.com/pgbackrest/pgbackrest/commit/67f35a88ca7c5ef702ce25c431bdaf8021328f45).
> 
> It's interesting that the switch statement did not complain that the type was not enum, even though the cases are enum. I'm not excited about just declaring everything as StringId because the enums provide some documentation and type safety in the cases where we are using the enum values.
>
So only certain compilers might complain? Because I did declare type as a StringId `StringId type = cfgOptionStrId(cfgOptType);` and the compiler I have didn't complain. 
maybe something in stringid.h
> 
> I like this idea better. How about `When assigning a StringId to an enum, it will be necessary to cast the StringId to the enum type if the enum contains all 32-bit values, since some compilers will complain about the implicit conversion without a cast. The enum will be 32-bit if all values of the enum are <= 0xffffffff.`
> 
> I not sure it I want to go with string length because some 6 character 5-bit strings would be 32-bit. In fact about 1/4 of them would be.
>
OK - you're wording looks good, then.

Reply:
> StringId type = cfgOptionStrId(cfgOptType);

This would always work because the type is the same as the return type. This will give an error depending on the compiler:
```
BackupType type = cfgOptionStrId(cfgOptType)
```
Note that this works just fine, but some compilers will warn which we make into an error while testing to be extra cautious.

Reply:
stringId.h updated at 4af2e6afff2363c.